src/storage/src/render/upsert.rs

```diff
@@ -317,7 +350,16 @@ where
                 )
                 .await
             {
-                Ok(_) => {}
+                Ok(_) => {
+                    if let Some(ts) = snapshot_upper.clone().into_option() {
+                        // As we shutdown, we could ostensibly get data from later than the
+                        // `resume_upper`, which we ignore above. We don't want our output capability to make
+                        // it further than the `resume_upper`.
```

Comment:
I don't think async operators need to care about shutdown anymore. Have you seen this being a problem?

Reply:
No I haven't, but without this token I _believe_ we could get stuck here, if the `input` is behind?